misc/python/materialize/mzbuild.py

```diff
@@ -314,26 +334,10 @@ def __init__(self, rd: RepositoryDetails, path: Path, config: Dict[str, Any]):
         self.strip = config.pop("strip", True)
         self.split_debuginfo = config.pop("split_debuginfo", False)
         self.extract = config.pop("extract", {})
-        self.rustflags = config.pop("rustflags", [])
```

Comment:
This is a regression, AFAICT. `rustflags` can be customized on a per-image basis. The antithesis image uses this in order to set a bunch of custom compiler flags. To preserve behavior, we need to keep this field here, and then have the common Cargo build command group each `ResolvedImage` by its `rustflags`, and only merge compilation when the `rustflags` match.

Alternatively, I'm comfortable with removing this feature to keep the code simple if we can also remove the `test/antithesis` directory.